### PR TITLE
added fix for volume and snapshot resource bugs

### PR DESCRIPTION
**Pull request by shenda1**
The PR is raised to fix volume and snapshot resource bugs

image

image

### CODE REVIEW

**Review by Krishnan-Priyanshu — APPROVED** [review]
lgtm

**Review by Krunal-Thakkar — APPROVED** [review]

**Review by Krunal-Thakkar — APPROVED** [review]

File: powerflex/snapshot_resource.go [hunk 1]
```diff
@@ -308,10 +308,15 @@ func (r *snapshotResource) Update(ctx context.Context, req resource.UpdateReques
 	}
 
 	// updating the retention in min if there is change in plan and state.
-	if (len(errMsg) == 0) && plan.DesiredRetention.ValueInt64() != state.DesiredRetention.ValueInt64() {
+	if plan.RetentionInMin.ValueString() != state.RetentionInMin.ValueString() {
```
Comment (AnikaAgiwal2711) [inline powerflex/snapshot_resource.go:311]: This change also solves the issue explained in TFM-988 ??

Comment (shenda1) [reply powerflex/snapshot_resource.go:311]: Yes..In the plan itself, desired retention and retention unit is getting converted to retention in minutes as API accepts retention in minutes

Comment (AnikaAgiwal2711) [reply powerflex/snapshot_resource.go:311]: ok. Thanks.

**Comment by AnikaAgiwal2711** [comment]
Looks good to me

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Pull request merged**